Week7/CurrentNews/app/src/main/java/com/florencenjeri/currentnews/ui/viewmodel/NewsViewModel.kt

```diff
@@ -5,9 +5,11 @@ import androidx.lifecycle.viewModelScope
 import com.florencenjeri.currentnews.database.NewsRepository
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import org.koin.core.KoinComponent
+import org.koin.core.inject
 
-class NewsViewModel(val newsRepository: NewsRepository) : ViewModel() {
-
+class NewsViewModel : ViewModel(), KoinComponent {
+    val newsRepository: NewsRepository by inject()
```

Comment:
This was fine in the constructor, and should've been there! Only inject activity/fragment dependencies!